src/Compilers/CSharp/Portable/Parser/QuickScanner.cs

```diff
@@ -152,7 +152,7 @@ private enum CharFlags : byte
                 (byte)QuickScanState.Done,                // Letter
                 (byte)QuickScanState.Number,              // Digit
                 (byte)QuickScanState.Done,                // Punct
-                (byte)QuickScanState.Done,                // Dot
+                (byte)QuickScanState.Bad,                 // Dot
```

Comment:
>(byte)QuickScanState.Bad,     [](start = 16, length = 29)

It would be good to add a comment explaining why this is the right value, what does it mean and what it makes other compiler to do.